### DISCUSSION TITLE
Support `indexed = "asc"/"desc"`

### DISCUSSION THREAD

**Discussion by stevefan1999-personal** [SeaQL/sea-orm#1647] [Ideas]
## Motivation

Right now it is impossible to write this:
```rs
#[derive(Clone, Debug, PartialEq, Eq, DeriveEntityModel)]
#[sea_orm(table_name = "kvstore")]
pub struct Model {
	#[sea_orm(primary_key, indexed = "asc", auto_increment = false)]
	pub key: Key,
	#[sea_orm(nullable = false)]
	pub value: Val,
}
```
I needed ascending index because I want to make hints explicit, despite that B-Tree index would sort by ascending order by default and is what my intended behavior, so if you switched index  types, that would very much not so.

## Proposed Solutions

Add a builder option `ColumnDef::index_order(order: IndexOrder) -> Self` to set the sort order for the index.

## Current Workarounds

There is no safe in-code workaround beside from dangerously iterating each index and check their logical name. Otherwise you can still alter the index in the database

**Comment by nickb937** [comment]
Index definitions can be fairly complex and perhaps should be defined standalone to the column.  The current system of defining the indexes lacks the expressiveness e.g. setting GIN or BRIN index type, or functional indexes (https://www.postgresql.org/docs/7.3/indexes-functional.html), or multi-column index.

I would think the way to go here is to have a IndexTrait that builds indexes using a builder-pattern, like the RelationTrait does for foreign keys.

**Comment by tyt2y3** [comment]
First of all the Entity definition does not do much, other than in create_table_from_entity.
You can create an index using SeaQuery in migrations.
Is there something missing?
